src/drivers/imu/analog_devices/adis16507/ADIS16507.cpp

```diff
@@ -88,96 +91,147 @@ void ADIS16507::exit_and_cleanup()
 	I2CSPIDriverBase::exit_and_cleanup();
 }
 
-void ADIS16507::print_status()
-{
-	I2CSPIDriverBase::print_status();
-
-	perf_print_counter(_reset_perf);
-	perf_print_counter(_bad_register_perf);
-	perf_print_counter(_bad_transfer_perf);
-	perf_print_counter(_perf_crc_bad);
-	perf_print_counter(_drdy_missed_perf);
-}
-
 int ADIS16507::probe()
 {
 	// Power-On Start-Up Time 310 ms
 	if (hrt_absolute_time() < 310_ms) {
-		PX4_WARN("required Power-On Start-Up Time 310 ms");
+		PX4_WARN("Required Power-On Start-Up Time 310 ms");
 	}
 
-	const uint16_t PROD_ID = RegisterRead(Register::PROD_ID);
+	const uint16_t id = RegisterRead(Register::PROD_ID);
 
-	if (PROD_ID != Product_identification) {
-		DEVICE_DEBUG("unexpected PROD_ID 0x%02x", PROD_ID);
+	if (id != Register::PROD_ID_EXPECTED) {
+		PX4_ERR("Unexpected PROD_ID 0x%02x", id);
 		return PX4_ERROR;
 	}
 
-	const uint16_t SERIAL_NUM = RegisterRead(Register::SERIAL_NUM);
-	const uint16_t FIRM_REV = RegisterRead(Register::FIRM_REV);
-	const uint16_t FIRM_DM = RegisterRead(Register::FIRM_DM);
-	const uint16_t FIRM_Y = RegisterRead(Register::FIRM_Y);
+	const uint16_t serial = RegisterRead(Register::SERIAL_NUM);
+	const uint16_t rev = RegisterRead(Register::FIRM_REV);
+	const uint16_t daymonth = RegisterRead(Register::FIRM_DM);
+	const uint16_t year = RegisterRead(Register::FIRM_Y);
 
-	PX4_INFO("Serial Number: 0x%X, Firmware revision: 0x%X Date: Y %X DM %X", SERIAL_NUM, FIRM_REV, FIRM_Y, FIRM_DM);
+	PX4_INFO("Serial Number: 0x%X, Firmware revision: 0x%X Date: Y %X DM %X", serial, rev, year, daymonth);
 
 	return PX4_OK;
 }
 
+bool ADIS16507::Configure()
+{
+	struct {
+		uint16_t reg;
+		uint16_t val;
+	} defaults[] = {
+		// Default 0x00C1
+		// - Change Data Ready polarity to active low
+		{ Register::MSC_CTRL, 0x00C0 },
+	};
+
+	// Write default register configuration
+	for (const auto &r : defaults) {
+		RegisterWrite(r.reg, r.val);
+	}
+
+	// We must wait for changes to apply
+	px4_usleep(5000);
```

Comment:
This one is kind of borderline, but anything more than this and we really shouldn't be blocking the bus.

Reply:
Yeah, it's probably longer than it needs to be. I didn't spent the time to tune the sleep. But it does need some time or the register check will fail. This is in Configure() and is only called after Reset() and typically there's nothing else on SPI6.

I'll see how short I can make it